### DISCUSSION TITLE
Transforming a mesh

### DISCUSSION THREAD

**Discussion by ofmetal** [elalish/manifold#1057] [Q&A] [answered]
I have a Manifold object and a transformation matrix which I want to apply:
```
Manifold m;
mat3x4 t;
m.Transform(t);
```
Then I extract vertex data after the transformation:
```
MeshGL6432& m2 = m.GetMeshGL6432();
for(auto a: m2.vertProperties) printf("%d\n", a);

```
But nothing is changed. All the vertex coordinates match to original m.
How can I get actually transformed coordinates?

**Comment by elalish** [comment, marked as answer]
Note that our API is functional - nearly every method is `const`. What you're missing is `m = m.Transform(t)`. We handle the eliding of copies internally.

**Reply by ofmetal** [reply]
That helped, thank you!
